source/install/config-ssl-http2-apache2.rst

```diff
@@ -0,0 +1,66 @@
+.. _config-ssl-http2-apache2:
+
+Configuring Apache2 with SSL and HTTP/2 (Unofficial)
+=====================================================
+
+.. important:: This unofficial guide is maintained by the Mattermost community and this deployment configuration is not yet officially supported by Mattermost, Inc. Community testing, feedback and improvements are welcome and greatly appreciated. You can `edit this page on GitHub <https://github.com/mattermost/docs/blob/master/source/install/config-ssl-http2-apache2.rst>`_.
+
+Once you've configured Apache2 as a proxy for your Mattermost Server, the easiest way to enable SSL on Apache2 is via Let's Encrypt and `Certbot <https://certbot.eff.org/#ubuntuxenial-apache>`_.
+
+Once installed, simply run:
+``$ certbot --apache`` and follow the straight forward guide. Afterwards you should find a new configuration file in ``/etc/apache2/sites-available`` which should follow the format: ``mysubdomain.mydomain.com-le-ssl.conf``.
+
+When opened, edit it to look something like the following:
+
+.. code-block:: apacheconf
+
+	<IfModule mod_ssl.c>
+	<VirtualHost *:443>
+		ServerName mysubdomain.mydomain.com
+		ServerAdmin hostmaster@mydomain.com
+		ProxyPreserveHost On
+
+		# setup the proxy
+		<Proxy *>
+		  Order allow,deny
+		  Allow from all
+		</Proxy>
+
+		RewriteEngine On
+		RewriteCond %{REQUEST_URI} ^/api/v3/users/websocket [NC,OR]
+		RewriteCond %{HTTP:UPGRADE} ^WebSocket$ [NC,OR]
+		RewriteCond %{HTTP:CONNECTION} ^Upgrade$ [NC]
+		RewriteRule .* ws://127.0.0.1:8065%{REQUEST_URI} [P,QSA,L]
+		RewriteCond %{DOCUMENT_ROOT}/%{REQUEST_FILENAME} !-f
+		RewriteRule .* http://127.0.0.1:8065%{REQUEST_URI} [P,QSA,L]
+
+		<Location /api/v3/users/websocket>
+		  Require all granted
+		  ProxyPass ws://127.0.0.1:8065/api/v3/users/websocket
+		  ProxyPassReverse ws://127.0.0.1:8065/api/v3/users/websocket
+		  ProxyPassReverseCookieDomain 127.0.0.1 mysubdomain.mydomain.com
+		</Location>
+
+		<Location />
+		  Require all granted
+		  ProxyPass http://127.0.0.1:8065/
+		  ProxyPassReverse http://127.0.0.1:8065/
+		  ProxyPassReverseCookieDomain 127.0.0.1 mysubdomain.mydomain.com
+		</Location>
+
+		# Generated by Certbot
+		SSLCertificateFile /etc/letsencrypt/live/mydomain.com/fullchain.pem
+		SSLCertificateKeyFile /etc/letsencrypt/live/mydomain.com/privkey.pem
+		Include /etc/letsencrypt/options-ssl-apache.conf
+	</VirtualHost>
+	</IfModule>
+
+1. Remember to Restart Apache2
+
+    - On Ubuntu 14.04 and RHEL 6: ``sudo service apache2 restart``
+    - On Ubuntu 16.04 and RHEL 7: ``sudo systemctl restart apache2``
+
+2. Test that the site is working, that sockets are working, and if you enabled HTTP redirect to HTTPS (during Certbot installation), that that is working.
+3. Lastly you should test your SSL integration with https://www.ssllabs.com/ssltest/index.html
+
+Using Certbot means that you shouldn't have to do anything in the configuration of Mattermost.
```

Comment:
Now this would be something worth investigating when more people working on it. As I used to Certbot to automatically install and configure Let's Encrypt, I wouldn't know.